libraries/lwc/src/basic-tests.js

```diff
@@ -0,0 +1,192 @@
+
+      toggler.click();
+      await Promise.resolve();
+      expectHasChildren(cmp.shadowRoot.querySelector('ce-with-children'));
+    });
+  });
+
+  describe('attributes and properties', function() {
+    it('will pass boolean data as either an attribute or a property', function() {
+      this.weight = 3;
+      const cmp = createComponentWithProperties();
+      scratch.appendChild(cmp);
+
+      const wc = cmp.shadowRoot.querySelector('ce-with-properties');
+      const data = wc.bool || wc.hasAttribute('bool');
+      expect(data).to.be.true;
+    });
+
+    it('will pass numeric data as either an attribute or a property', function() {
+      this.weight = 3;
+      const cmp = createComponentWithProperties();
+      scratch.appendChild(cmp);
+
+      const wc = cmp.shadowRoot.querySelector('ce-with-properties');
+      const data = wc.num || wc.getAttribute('num');
+      expect(parseInt(data, 10)).to.eql(42);
+    });
+
+    it('will pass string data as either an attribute or a property', function() {
+      this.weight = 3;
+      const cmp = createComponentWithProperties();
+      scratch.appendChild(cmp);
+
+      const wc = cmp.shadowRoot.querySelector('ce-with-properties');
+      const data = wc.str || wc.getAttribute('str');
+      expect(data).to.eql('lwc');
+    });
+
+    /*
+    it('will set boolean attributes on a Custom Element that has not already been defined and upgraded', function() {
+      const cmp = ComponentWithUnregistered();
+      scratch.appendChild(cmp);
+      const wc = cmp.shadowRoot.querySelector('ce-unregistered');
+      expect(wc.hasAttribute('bool')).to.be.true;
+    });
+
+    it('will set numeric attributes on a Custom Element that has not already been defined and upgraded', function() {
+      const cmp = ComponentWithUnregistered();
+      scratch.appendChild(cmp);
+      const wc = cmp.shadowRoot.querySelector('ce-unregistered');
+      expect(wc.getAttribute('num')).to.eql('42');
+    });
+
+    it('will set string attributes on a Custom Element that has not already been defined and upgraded', function() {
+      const cmp = ComponentWithUnregistered();
+      scratch.appendChild(cmp);
+      const wc = cmp.shadowRoot.querySelector('ce-unregistered');
+      expect(wc.getAttribute('str')).to.eql('lwc');
+    });
+
+    it('will set array properties on a Custom Element that has not already been defined and upgraded', function() {
+      const cmp = ComponentWithUnregistered();
+      scratch.appendChild(cmp);
+      const wc = cmp.shadowRoot.querySelector('ce-unregistered');
+      expect(wc.arr).to.eql(['L', 'W', 'C']);
+    });
+
+    it('will set object properties on a Custom Element that has not already been defined and upgraded', function() {
+      const cmp = ComponentWithUnregistered();
+      scratch.appendChild(cmp);
+      const wc = cmp.shadowRoot.querySelector('ce-unregistered');
+      expect(wc.obj).to.eql({ org: 'salesforce', repo: 'lwc' });
+    });
+    */
```

Comment:
Do we need this commented-out code?

Reply:
I implemented these before realizing that the other libraries had commented them out so I went ahead and did the same. I'll just remove it.